chacra/async/__init__.py

```diff
@@ -0,0 +1,117 @@
+import pecan
+from celery import Celery
+import celery
+from datetime import timedelta
+from chacra import models
+from chacra.util import repo_directory
+import os
+import logging
+logger = logging.getLogger(__name__)
+
+
+def get_pecan_config():
+    try:
+        os.environ['PECAN_CONFIG']
+    except KeyError:
+        here = os.path.abspath(os.path.dirname(__file__))
+        config_path = os.path.abspath(os.path.join(here, '../config/config.py'))
+        return config_path
+
+
+pecan.configuration.set_config(get_pecan_config(), overwrite=True)
+# FIXME: this needs to be configurable. Also, not sure if this is the right way
+# to configure Celery for multiple tasks running in a single service.
+# TODO: Investigate if this could be `chacra.async.rpm` and
+# `chacra.async.debian` or if it doesn't matter
+app = Celery('chacra.async', broker='amqp://guest@localhost//')
+
+models.init_model()
+
+
+class SQLATask(celery.Task):
+    """An abstract Celery Task that ensures that the connection the the
+    database is closed on task completion"""
+    abstract = True
+
+    def after_return(self, status, retval, task_id, args, kwargs, einfo):
+        return
+        models.remove()
+
+
+@app.task(base=SQLATask)
+def poll_repos():
+    """
+    Poll the repository objects that need to be updated and call the tasks
+    that can create (or update) repositories with that information
+
+    """
+    logger.info('polling repos....')
+    from chacra.models import Repo
+    for r in Repo.query.all():
+        logger.info(r)
+        logger.info(r.binaries)
+        if r.needs_update:
+            logger.info("repo %s needs to be updated/created", r)
+            if r.binaries[0].name.endswith('rpm'):
+                create_repo.delay([r.id])
+
+    logger.info('completed repo polling')
+
+
+@app.task(base=SQLATask)
+def create_repo(repo_ids):
+    """
+    Go create or update repositories with specific IDs.
+    """
+    from chacra.models import Repo
```

Comment:
I assume this is for the circular import you mentioned.  You're already importing ``models`` above, could you use ``models.Repo`` in the task and avoid this import?

Or maybe the trouble is because we're import ``chacra.models.repo.Repo`` into ``chacra.models`` and shouldn't be?